File: db4o.cs/src/com/db4o/cs/internal/config/Db4oClientServerLegacyConfigurationBridge.java
/* Copyright (C) 2009 Versant Inc.   http://www.db4o.com */
package com.db4o.cs.internal.config;

import com.db4o.config.*;
import com.db4o.cs.config.*;
import com.db4o.internal.*;
import com.db4o.internal.config.*;

public class Db4oClientServerLegacyConfigurationBridge {

	public static ClientConfiguration asClientConfiguration(Configuration config) {
		return new ClientConfigurationImpl((Config4Impl) config);
	}

	public static ServerConfiguration asServerConfiguration(Configuration config) {
		return new ServerConfigurationImpl((Config4Impl) config);
	}

	public static Config4Impl asLegacy(Object config) {
		return ((LegacyConfigurationProvider)config).legacy();
	}

	public static NetworkingConfiguration asNetworkingConfiguration(Configuration config) {
		return asServerConfiguration(config).networking();
	}

}

File: db4o.net/Db4objects.Db4o.CS/Db4objects.Db4o.CS/CS/Internal/Config/Db4oClientServerLegacyConfigurationBridge.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.CS.Config;
using Db4objects.Db4o.CS.Internal.Config;
using Db4objects.Db4o.Config;
using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Config;

namespace Db4objects.Db4o.CS.Internal.Config
{
	public class Db4oClientServerLegacyConfigurationBridge
	{
		public static IClientConfiguration AsClientConfiguration(IConfiguration config)
		{
			return new ClientConfigurationImpl((Config4Impl)config);
		}

		public static IServerConfiguration AsServerConfiguration(IConfiguration config)
		{
			return new ServerConfigurationImpl((Config4Impl)config);
		}

		public static Config4Impl AsLegacy(object config)
		{
			return ((ILegacyConfigurationProvider)config).Legacy();
		}

		public static INetworkingConfiguration AsNetworkingConfiguration(IConfiguration config
			)
		{
			return AsServerConfiguration(config).Networking;
		}
	}
}
